liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/jsp/CompileJspMojo.java

```diff
@@ -109,7 +110,18 @@ protected void doExecute() throws Exception {
             compile.setJspVersion(jspVersion);
         }
 
-        // TODO do we need to add features?
+        if(initialize()) {
+            Set<String> installedFeatures = getInstalledFeatures();
+
+            //Removing jsp features at it is already set at this point 
+            installedFeatures.remove("jsp-2.3");
+            installedFeatures.remove("jsp-2.2");
```

Comment:
When someone runs compile-jsp, the server.xml that gets created already includes the `jsp-2.x` feature from what is specified in the last snippet. We already have the `jspVersion` parameter they could specify in the pom if they want jsp-2.2. 

Here, I'm just removing the jsp-feature from the feature list so there aren't duplicates. Without this snippet,if someone were to specify `<feature>jsp-2.X</feature>` in their server.xml or POM and then run `liberty:compile-jsp`, the resulting server.xml for the compile-jsp server would have the jsp-feature listed twice.

Reply:
I was just suggesting that if they bothered to include a jsp feature in their server.xml, it seems like we should honor that (not if it conflicted with the pom.xml parameter though). I understood why you were removing them.

Reply:
Ah I get you now. This would require us to check if they specified a jsp-feature in the pom, and then check that version. This also opens up the question as to what would take precedence if they specify a different version using the pom parameter `jspVersion`?